### PR TITLE
feat(vmguests): adding vm tools related metrics (status, version, version status)

### LINKED ISSUE #71: Add VM metric to indicate vm tools status
This will be very helpful to ensure vmtools is running on all VMs as required, perhaps we can even include the version.

We're actually hitting an issue form time to time where our terraform plan/apply stalls because some vmtools crashed on some VMs and it's a bit tricky to hunt down which one(s), this would make our lives much easier.

an example is available [here](https://github.com/vmware/pyvmomi-community-samples/blob/master/samples/list_vmwaretools_status.py)

I'll try to prepare a PR for this (if no objections), once we move forward with the more heavier changes in the pipe.

### PR DESCRIPTION
**Pull request by dannyk81**
fixes #71 

let me know what you guys think, I've been testing this in my cluster and works quite well.

examples:
```
vmware_vm_guest_tools_running_status{cluster_name="***",dc_name="***",environment="***",host_name="***",instance="***",job="vmware-exporter",tools_status="toolsOk",vm_name="***"} | 1

vmware_vm_guest_tools_version{cluster_name="***",dc_name="***",environment="***",host_name="***",instance="***",job="vmware-exporter",tools_version="10304",vm_name="***"} | 1

vmware_vm_guest_tools_version_status{cluster_name="***",dc_name="***",environment="***",host_name="***",instance="***",job="vmware-exporter",tools_version_status="guestToolsUnmanaged",vm_name="***"} | 1
```

### CODE REVIEW

**Comment by pryorda** [comment]
Can you add tests for this in the test files? Otherwise lgtm.

**Comment by dannyk81** [comment]
yeah... working on tests

**Comment by pryorda** [comment]
Ha, It just updated to [WIP]. My bad :)

**Comment by dannyk81** [comment]
tests added and squashed

/cc @pryorda

File: tests/unit/test_vmware_exporter.py [hunk 1]
```diff
@@ -116,6 +119,34 @@ def test_collect_vms():
...
+        'tools_version': '10336',
+    }
+    assert metrics['vmware_vm_guest_tools_version'].samples[0][2] == 1.0
+
+    assert metrics['vmware_vm_guest_tools_version_status'].samples[0][1] == {
+        'vm_name': 'vm-1',
+        'host_name': 'host-1',
+        'cluster_name': 'cluster-1',
+        'dc_name': 'dc',
+        'tools_version_status': 'guestToolsUnmanaged',
```
Comment (pryorda) [inline tests/unit/test_vmware_exporter.py]: tools_status or tools_version_status?

Comment (pryorda) [reply tests/unit/test_vmware_exporter.py]: If the api reveals it as tools_version_status. I say leave it.

Comment (dannyk81) [reply tests/unit/test_vmware_exporter.py]: There are 3 different attributes:
1) `tools_running_status` - indicates if the tools is running or not (i.e. `toolsOk`)
2) `tools_version` - the version of the installed tools
3) `tools_version_status` - whether it is managed on unmanaged deployed (i.e. if it's official vmware tools or the open-vm-tools)

I'm open to changing the label/metric names if you think it's confusing, but seems to be quite aligned with the api.

We could potentially even merge this into a single metric, but to be able to safe guard each value, I suppose it's better to keep them separate.

File: tests/unit/test_vmware_exporter.py [hunk 2]
```diff
@@ -116,6 +119,34 @@ def test_collect_vms():
     }
     assert metrics['vmware_vm_guest_disk_capacity'].samples[0][2] == 100
 
+    # VM tools info (vmguest)
+    assert metrics['vmware_vm_guest_tools_running_status'].samples[0][1] == {
+        'vm_name': 'vm-1',
+        'host_name': 'host-1',
+        'cluster_name': 'cluster-1',
+        'dc_name': 'dc',
+        'tools_status': 'toolsOk',
```
Comment (dannyk81) [inline tests/unit/test_vmware_exporter.py]: maybe just call this label `status`?

File: tests/unit/test_vmware_exporter.py (hunk 1 above)
Comment (pryorda) [reply tests/unit/test_vmware_exporter.py]: I'd rather have it match the api tbh

File: tests/unit/test_vmware_exporter.py (hunk 2 above)
Comment (pryorda) [reply tests/unit/test_vmware_exporter.py]: I say just have it match the api. I don't like interpreting external apis in our own way. Creates a level of abstraction thats is not needed.

**Comment by pryorda** [comment]
Feel free to rebase and merge. It will auto create the release and stuffs.

**Comment by pryorda** [comment]
That was easy

**Pull request merged**